bitnami/discourse/templates/_helpers.tpl

```diff
@@ -276,12 +315,29 @@ Return the Redis Port
 {{- end -}}
 
 {{/*
-Return the Redis Secret
+Return true if a secret object for Redis should be created
+*/}}
+{{- define "discourse.redisCreateSecret" -}}
+{{- if or (and .Values.redis.enabled .Values.redis.existingSecret) (and (not .Values.redis.enabled) .Values.externalRedis.existingSecret) }}
+    {{- false -}}
+{{- else -}}
+    {{- true -}}
+{{- end -}}
+{{- end -}}
+
+{{/*
+Return the Redis secret name
 */}}
 {{- define "discourse.redisSecretName" -}}
```

Comment:
As I have mention, I'd rather use `discourse.secretName` rather than `discourse.discourseSecretName`. The chart uses this naming pattern `discourse.redisSecretName`, it is okay if you leave `database` and `redis` as it is, but I'd definitely change the `discourse` one to `discourse.secretName`.